### ISSUE TITLE
keras pipe mode - Invalid header crc error

### ISSUE THREAD

**Issue by jamiekang** [aws/sagemaker-tensorflow-extensions#72]
Hi,
when I run the pipe mode part of [keras_script_mode_pipe_mode_horovod](https://github.com/awslabs/amazon-sagemaker-examples/blob/master/sagemaker-python-sdk/keras_script_mode_pipe_mode_horovod/tensorflow_keras_CIFAR10.ipynb) example, I find this error in the Cloudwatch console.
`tensorflow.python.framework.errors_impl.InternalError: Invalid header crc`

![Screen Shot 2020-04-17 at 11 42 35 AM](https://user-images.githubusercontent.com/16663823/79624814-5ff52900-815f-11ea-86e5-fe77e56be4a6.png)

It seems related with tensorflow extensions: https://github.com/aws/sagemaker-tensorflow-extensions/blob/master/src/pipemode_op/RecordReader/TFRecordReader.cpp

It does not happen in other parts of the notebook, e.g. local mode, cloud, or horovod.
I could replicate this problem with several SageMaker notebooks in multiple regions.

How can I work around this?

**Comment by laurenyu** [comment]
sorry for the delayed response.

I started a SageMaker Notebook Instance, ran the notebook you linked to, and the pipe mode training job succeeded for me. Could you list more details about your environment (e.g. notebook instance type, notebook kernel, etc.)?

**Comment by jamiekang** [comment]
Hi,
my notebook instance is ml.p2.xlarge in N.Virginia.
I am running conda_tensorflow_p36 kernel.
I got the same error today (seems not always).

**Comment by jamiekang** [comment]
Hi Lauren, 
I found what caused the issue.
For some reason, I had some garbage files under `/data/train` along with the data, `train.tfrecords`.

I specified the whole filename instead of just the folder name as this:
`remote_inputs = {'train' : dataset_location+'/train/train.tfrecords', 'validation' : dataset_location+'/validation', 'eval' : dataset_location+'/eval'}`

It's because  there is no way to validate the streaming data in pipe mode.

I am resolving this issue.
